app/src/XRayServer.cpp

```diff
@@ -0,0 +1,117 @@
+#include "XRayServer.h"
+
+#include <jsonrpc/net/ServerTransport.h>
+#include <jsonrpc/rpc/Dispatcher.h
+
+#include <string>
+
+using namespace jsonrpc;
+
+namespace xray
+{
+
+struct xray::GetValue
+{
+	GetValue(double& value)
+		: _value(value)
+	{}
+	rpc::Json call(const rpc::Json& data) override
+	{
+		return _value;
+	}
+
+private:
+	double& _value;
+};
+
+struct xray::SetValue
+{
+	SetValue(double& value) : _value(value) {}
+	rpc::Json call(const rpc::Json& data) override
+	{
+		_value = data;
+		return {};
+	}
+	private:
+	double& _value;
+};
+
+struct xray::setTubeVoltage
+{
+	rpc::Json call(const rpc::Json& data) override
+	{
+		//TODO: error handling (check if Voltage is in range)
+		if (data.is_array())
+		{
+			std::array<double, 1> par = data;
+			voltage                   = par[0];
+		}
+		else if (data.is_object())
```

Comment:
with what parameters shall the method be called?

1. `[42.0, ]`
2. `{voltage: 42.0}`
3. `42.0`

I think we should only support one version, preferably the simplest.